### PR TITLE
fix: missing root path for logout

### PR DESCRIPTION
**Pull request by RogerHYang**
> [!NOTE]
> Introduce `prepend_root_path`/`get_root_path` and use them to correctly build redirects, password reset links, and UI basename when app is mounted under a subpath.
> 
> - **Utilities**:
>   - Add `phoenix.server.utils` with `get_root_path(scope)` and `prepend_root_path(scope, path)` for normalized root path handling.
> - **Auth** (`src/phoenix/server/api/routers/auth.py`):
>   - Use `prepend_root_path` for logout redirect and password reset link construction (replace `Path`/`get_env_host_root_path`).
> - **OAuth2** (`src/phoenix/server/api/routers/oauth2.py`):
>   - Replace custom root-path helpers with `prepend_root_path`/`get_root_path`.
>   - Apply to login error redirects and token redirect URLs; update `_append_root_path_if_exists` to use `get_root_path`.
> - **App/UI** (`src/phoenix/server/app.py`):
>   - Use `prepend_root_path` for auto-login redirect; use `get_root_path` for template `basename`; remove local basename sanitizer.
> - **Tests**:
>   - Add unit tests for `get_root_path` and `prepend_root_path` covering normalization and edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ec19e5b7d6765083aea047e9217fe1502dc5587b. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/phoenix/server/api/routers/utils.py [hunk 1]
```diff
@@ -19,3 +21,73 @@ def from_iso_format(value: Optional[str]) -> Optional[datetime]:
...
+        otherwise just the normalized path.
+
+    Examples:
+        With root_path="/api/v1":
+        - prepend_root_path_if_exists(request, "/login") -> "/api/v1/login"
+        - prepend_root_path_if_exists(request, "login") -> "/api/v1/login"
+
+        With no root_path:
+        - prepend_root_path_if_exists(request, "/login") -> "/login"
+        - prepend_root_path_if_exists(request, "login") -> "/login"
```
Comment (mikeldking) [inline src/phoenix/server/api/routers/utils.py]: this is a bit of a confusing example with `/api/v1` - better to make a realistic example
```suggestion
    Examples:
        With root_path="/apps/phoenix":
        - prepend_root_path_if_exists(request, "/login") -> "/apps/phoenix/login"
        - prepend_root_path_if_exists(request, "login") -> "/apps/phoenix/login"

        With no root_path:
        - prepend_root_path_if_exists(request, "/login") -> "/login"
        - prepend_root_path_if_exists(request, "login") -> "/login"
```

**Review by mikeldking — APPROVED** [review]
